web-bundle/src/main/java/com/graphhopper/customspeeds/CustomSpeedsVehicle.java

```diff
@@ -58,12 +60,17 @@ public static boolean validateCustomSpeeds(String customVehicleName, ImmutableMa
         // Validation of each custom speed's direction is done separately in the vehicle access parsers, via CustomSpeedUtils.validateCustomSpeedDirection()
         VehicleType baseVehicleType = CustomSpeedsVehicle.getBaseVehicleType(customVehicleName);
 
-        // wrap filterValues result in a HashMap to turn live, lazy view into a traditional map
-        Map<Long, DirectedSpeeds> invalidSpeeds = new HashMap<>(Maps.filterValues(osmWayIdToDirectedMaxSpeeds,
-                directedSpeeds -> directedSpeeds.getSpeedsByDirection().values().stream().anyMatch(speed -> speed < 0 || speed > baseVehicleType.getMaxValidSpeed())));
-        if (!invalidSpeeds.isEmpty()) {
+        // wrap transformValues result in a HashMap to turn live, lazy view into a traditional map
+        Map<Long, Map<SpeedsDirection, Double>> osmWayIdToInvalidSpeeds = new HashMap<>(Maps.transformValues(osmWayIdToDirectedMaxSpeeds,
+                directedSpeeds -> directedSpeeds.speedsByDirection.entrySet().stream()
+                        .filter(entry -> entry.getValue() < 0 || entry.getValue() > baseVehicleType.getMaxValidSpeed())
+                        .collect(Collectors.toUnmodifiableMap(Map.Entry::getKey, Map.Entry::getValue)
+        )));
+        osmWayIdToInvalidSpeeds = Maps.filterValues(osmWayIdToInvalidSpeeds, invalidSpeedsMap -> !invalidSpeedsMap.isEmpty());
```

Comment:
fyi if you're not aware, these guava maps functions like `Maps.filterValues` and `Maps.transformValues` generate a lazy, live view of the map, meaning the predicate is continually evaluated even after the map is created ([docs](https://guava.dev/releases/19.0/api/docs/com/google/common/collect/Maps.html#filterValues(java.util.Map,%20com.google.common.base.Predicate))). probably doesn't matter here much, but i usually prefer to wrap the view in a new map so it's no longer live and the perf is more standard. realize i'm being paranoid, but i've been bit before by long chains of filterValues/transformValues which caused a big perf degradation 🙃

Reply:
Good point! I'd noticed your comment just above here + elsewhere about this and I agree it's a safer approach in most cases to force a non-lazy evaluation of the maps resulting from these functions when possible. Made the update here, and checked all other cases of `filterValues` and `transformValues`; the only case still not wrapped in a standard map instantiation is [here](https://github.com/replicahq/graphhopper/pull/171/files#diff-05fbfecbd80955330167a2e3f4d60e3047ff0b0de220a28e242525af5fe0123bR144-R146), but I think it's justifiable as the only thing we're doing with the result is immediately passing it into `ImmutableMap.copyOf()`